jdaviz/app.py

```diff
@@ -1610,19 +1606,15 @@ def update_viewer_reference_name(
             self._viewer_store[new_reference] = self._viewer_store.pop(old_id)
             self.state.viewer_icons[new_reference] = self.state.viewer_icons.pop(old_id)
 
-        # Update the viewer name attributes on the helper. First check if
-        # an attr for this viewer reference name is stored on the helper:
-        if old_reference_helper_attr is None:
-            old_viewer_ref_attrs = [
-                attr for attr in dir(self._jdaviz_helper)
-                if old_reference.replace('-', '_') in attr
-            ]
-            if old_viewer_ref_attrs:
-                old_reference_helper_attr = old_viewer_ref_attrs[0]
-
-        # if there is a detectable attr to update, update it:
-        if old_reference_helper_attr:
-            setattr(self._jdaviz_helper, old_reference_helper_attr, new_reference)
+        # update the viewer name attributes on the helper:
+        old_viewer_ref_attrs = [
+            attr for attr in dir(self._jdaviz_helper)
+            if attr.startswith('_default_') and
+            getattr(self._jdaviz_helper, attr) == old_reference
+        ]
+        if old_viewer_ref_attrs:
+            # if there is an attr to update, update it:
+            setattr(self._jdaviz_helper, old_viewer_ref_attrs[0], new_reference)
```

Comment:
why not just loop over `old_viewer_ref_attrs`?  If its empty, nothing will happen, and then if for any reason there is more than one match, all of them will be updated?

Reply:
I guess that depends on whether we want to assume and/or check if viewer names are unique. Thoughts